rotkehlchen/accounting/structures/evm_event.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Optional
+
+from rotkehlchen.accounting.structures.base import (
+    HISTORY_BASE_ENTRY_DB_TUPLE_WRITE,
+    HISTORY_EVENT_DB_TUPLE_READ,
+    HistoryBaseEntry,
+)
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_optional
+from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+EVM_EVENT_FIELDS = (
+    Optional[str],  # protocol
+    Optional[str],  # product
+    Optional[str],  # address
+    Optional[str],  # extra_data
+)
+
+
+EVM_EVENT_DB_TUPLE_WRITE = HISTORY_BASE_ENTRY_DB_TUPLE_WRITE + EVM_EVENT_FIELDS
+
+EVM_EVENT_DB_TUPLE_READ = HISTORY_EVENT_DB_TUPLE_READ + EVM_EVENT_FIELDS
+
+
+SUB_SWAPS_DETAILS = 'sub_swaps'
+LIQUITY_STAKING_DETAILS = 'liquity_staking'
+
+ALL_DETAILS_KEYS = {
+    SUB_SWAPS_DETAILS,
+    LIQUITY_STAKING_DETAILS,
+}
+
+
+def get_tx_event_type_identifier(event_type: HistoryEventType, event_subtype: HistoryEventSubType, protocol: str) -> str:  # noqa: E501
+    return str(event_type) + '__' + str(event_subtype) + '__' + protocol
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class EvmEvent(HistoryBaseEntry):
+    # Used to mark the protocol name, for example curve or liquity.
+    protocol: Optional[str] = None
+
+    # For example if we are interacting with a pool, staking contract or others.
+    # This will help when filtering the events adding easier granularity to the searches.
+    product: Optional[str] = None
+
+    # If we are working with evm information this would be the address of the contract.
+    # This would help to filter by older versions or limit searches to certain subsets of contracts.  # noqa: E501
+    # For example this would help filtering interactions with curve gauges.
+    address: Optional[ChecksumEvmAddress] = None
+
+    # Contains event specific extra data. Optional, only for events that need to keep
+    # extra information such as the CDP ID of a makerdao vault etc.
+    extra_data: Optional[dict[str, Any]] = None
+
+    def serialize_evm_event_for_db(self) -> EVM_EVENT_DB_TUPLE_WRITE:
+        return (
+            self.protocol,
+            self.product,
+            self.address,
+            json.dumps(self.extra_data) if self.extra_data else None,
+        )
+
+    def serialize(self) -> dict[str, Any]:
+        return super().serialize() | {
+            'protocol': self.protocol,
+            'product': self.product,
+            'address': self.address,
+            'extra_data': self.extra_data,
+        }
+
+    def serialize_without_extra_data(self) -> dict[str, Any]:
+        result = self.serialize()
+        result.pop('extra_data')
+        return result
+
+    @classmethod
+    def deserialize_evm_event_from_db(cls, entry: EVM_EVENT_DB_TUPLE_READ) -> 'EvmEvent':
```

Comment:
The problem is that if we use `deserialize_from_db` as a name here, it will break inheritance rules (the method in `BaseHistoryEntry` and this one would have different input arguments types).